File: nativelib/src/main/scala/scala/scalanative/native/utime.scala
package scala.scalanative
package native

@extern
object utime {
  // TODO: Where should we put those recurrent types?
  import stat.time_t
  type utimbuf = CStruct2[time_t, // actime
                          time_t] // modtime

  @name("scalanative_utime")
  def utime(path: CString, times: Ptr[utimbuf]): CInt = extern
}

File: posixlib/src/main/resources/scala-native/utime.c
#if defined(__unix__) || defined(__unix) || defined(unix) ||                   \
    (defined(__APPLE__) && defined(__MACH__))
#include <utime.h>

int scalanative_utime(char *path, struct utimbuf *times) {
    return utime(path, times);
}
#endif // Unix or Mac OS
